events/generator/generator.go

```diff
@@ -0,0 +1,42 @@
+package generator
+
+import (
+	"chainguard.dev/sdk/civil"
+)
+
+const GenerateVulnReportEventType = "dev.chainguard.vulnreport.generate.v1"
+
+type Error struct {
+	Status  int    `json:"status"`
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+// GenerateVulnReportEvent is an event used to trigger the generation of scan reports on-demand
+type GenerateVulnReportEvent struct {
+	// RepoID identifies the UIDP of the repository being pushed
+	RepoID string `json:"repo_id"`
+
+	// Digest holds the digest being pushed.
+	// Digest will hold the sha256 of the content being pushed, whether that is
+	// a blob or a manifest.
+	Digest string `json:"digest"`
+
+	// Type determines whether the object being pushed is a manifest or blob.
+	Type string `json:"type"`
+
+	// When holds when the push occurred.
+	When civil.DateTime `json:"when"`
+
+	// Location holds the detected approximate location of the client who pushed.
+	// For example, "ColumbusOHUS" or "Minato City13JP".
+	Location string `json:"location"`
+
+	// UserAgent holds the user-agent of the client who pushed.
+	UserAgent string `json:"user_agent"`
+
+	Error *Error `json:"error,omitempty"`
+
+	// RepoManifestDigestID identifies the UIDP of the repo_manifest_digest being pushed
+	RepoManifestDigestID string `json:"repo_manifest_digest_id"`
```

Comment:
These are the output values? Should these be separated into a separate type?

somewhat related - you may find https://github.com/googleapis/googleapis/blob/2610c7a8f3c362f1ef54cb0bb636ded47422eec8/google/longrunning/operations.proto#L128-L162, https://google.aip.dev/151 interesting - lots of similarities here.

Reply:
I am not going to use it so we can omit it.

Reply:
I decided to simplify the type itself.